### DISCUSSION TITLE
Any consideration when choosing external etcd or internal etcd?

### DISCUSSION THREAD

**Discussion by Reidddddd** [milvus-io/milvus#32863] [Q&A and General discussion]
Having no experience in milvus prod environment yet, don't know about the pros and cons

Can anyone share your opinions / experiences 

likewise for object store and message queue (suggest to use internal or external?)

highly appreciated!

**Comment by yhmo** [comment]
Highly recommend using Milvus cluster in prod environment.
Inernal etcd is only for Milvus standalone. It is impossible to use internal etcd in Milvus cluster.
Object store: minio, aws, gcp, aliyun. Depends on your preference.
Message queue: kafka, Pulsar. Depends on your preference.

**Reply by Reidddddd** [reply]
Hi, what about the mq and os, is it recommended to use inCluster or external mode

**Reply by yhmo** [reply]
Only standalone can use local storage as object store. 
Only standalone can use local storage as message queue.
Standalone and local object store, local message queue is not recommended to be used in prod environment.

If you just try standalone for personal test, no much concern, it doesn't matter what os/mq is used.
